app/src/redux-resources/analytics/hooks/useCameraAnalytics.ts

```diff
@@ -0,0 +1,186 @@
+import {
+  ANALYTICS_CAMERA_ENABLEMENT_KIND,
+  ANALYTICS_CAMERA_SETTINGS_KIND,
+  ANALYTICS_IMAGE_CAPTURE_KIND,
+  ANALYTICS_LIVE_FEED_KIND,
+  ANALYTICS_PHOTO_ACCESS,
+  useTrackEvent,
+} from '/app/redux/analytics'
+
+import type { RobotType } from '@opentrons/shared-data'
+import type {
+  SOURCE_ROBOT_SETTINGS,
+  SOURCE_RUN_RECORD,
+} from '/app/redux/analytics'
+
+export interface CameraAnalyticsParams {
+  source: typeof SOURCE_RUN_RECORD | typeof SOURCE_ROBOT_SETTINGS
+  robotType?: RobotType
```

Comment:
```suggestion
  robotType: RobotType
```
Think this should be required. We will always know the robot type. Either we are on a page that can derive this info or we are on the ODD, in which case we know we are on the flex.